tests/filecheck/dialects/riscv/riscv_assembly_emission.mlir

```diff
@@ -1,178 +1,187 @@
+    %slt = "riscv.slt"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: slt j2, j2, j1
+    %sltu = "riscv.sltu"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sltu j2, j2, j1
+    %and = "riscv.and"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: and j2, j2, j1
+    %or = "riscv.or"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: or j2, j2, j1
+    %xor = "riscv.xor"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: xor j2, j2, j1
+    %sll = "riscv.sll"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sll j2, j2, j1
+    %srl = "riscv.srl"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: srl j2, j2, j1
+    %sub = "riscv.sub"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sub j2, j2, j1
+    %sra = "riscv.sra"(%2, %1) : (!riscv.reg<j2>, !riscv.reg<j1>) -> !riscv.reg<j2>
+    // CHECK-NEXT: sra j2, j2, j1
+    "riscv.nop"() : () -> ()
+    // CHECK-NEXT: nop
+
+    // RV32I/RV64I: 2.5 Control Transfer Instructions
+
+    // Unconditional Branch Instructions
+    "riscv.jal"() {"immediate" = 1 : i32} : () -> ()
+    // CHECK-NEXT: jal 1
+    "riscv.jal"() {"immediate" = 1 : i32, "rd" = !riscv.reg<s0>} : () -> ()
+    // CHECK-NEXT: jal s0, 1
+    "riscv.jal"() {"immediate" = #riscv.label<"label">} : () -> ()
+    // CHECK-NEXT: jal label
+
+    "riscv.j"() {"immediate" = 1 : i32, "rd" = !riscv.reg<zero>} : () -> ()
+    // CHECK-NEXT: j 1
+    "riscv.j"() {"immediate" = #riscv.label<"label">, "rd" = !riscv.reg<zero>} : () -> ()
+    // CHECK-NEXT: j label
+
+    "riscv.jalr"(%0) {"immediate" = 1 : i32}: (!riscv.reg<zero>) -> ()
+    // CHECK-NEXT: jalr zero, 1
+    "riscv.jalr"(%0) {"immediate" = 1 : i32, "rd" = !riscv.reg<j0>} : (!riscv.reg<zero>) -> ()
+    // CHECK-NEXT: jalr j0, zero, 1
+    "riscv.jalr"(%0) {"immediate" = #riscv.label<"label">} : (!riscv.reg<zero>) -> ()
+    // CHECK-NEXT: jalr zero, label
+
+    "riscv.ret"() : () -> ()
+    // CHECK-NEXT: ret
+    ^0(%b00 : !riscv.reg<>, %b01 : !riscv.reg<>):
+
+
+    // Conditional Branch Instructions
+    "riscv.beq"(%2, %1) {"offset" = 1 : i32}: (!riscv.reg<j2>, !riscv.reg<j1>) -> ()
+    // CHECK-NEXT: beq j2, j1, 1
+    "riscv.bne"(%2, %1) {"offset" = 1 : i32}: (!riscv.reg<j2>, !riscv.reg<j1>) -> ()
+    // CHECK-NEXT: bne j2, j1, 1
+    "riscv.blt"(%2, %1) {"offset" = 1 : i32}: (!riscv.reg<j2>, !riscv.reg<j1>) -> ()
+    // CHECK-NEXT: blt j2, j1, 1
+    "riscv.bge"(%2, %1) {"offset" = 1 : i32}: (!riscv.reg<j2>, !riscv.reg<j1>) -> ()
+    // CHECK-NEXT: bge j2, j1, 1
+    "riscv.bltu"(%2, %1) {"offset" = 1 : i32}: (!riscv.reg<j2>, !riscv.reg<j1>) -> ()
+    // CHECK-NEXT: bltu j2, j1, 1
+    "riscv.bgeu"(%2, %1) {"offset" = 1 : i32}: (!riscv.reg<j2>, !riscv.reg<j1>) -> ()
+    // CHECK-NEXT: bgeu j2, j1, 1
```

Comment:
Similarly here, shouldn't these be jumping to other blocks?

Reply:
That's in the next PR, this PR just sets the scene for that change